ui/app/lib/kv-breadcrumbs.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+function pathIsDirectory(pathToSecret) {
+  // This regex only checks for / at the end of the string. ex: boop/ === true, boop/bop === false;
+  return pathToSecret ? !!pathToSecret.match(/\/$/) : false;
+}
+
+function pathIsFromDirectory(path) {
+  // This regex just looks for a / anywhere in the path. ex: boop/ === true, boop/bop === true;
+  return path ? !!path.match(/\//) : false;
+}
+
+function breadcrumbsForDirectory(path) {
+  // path === "beep/boop/"
+  const pathAsArray = path.split('/').filter((path) => path);
+  const modelIdArray = pathAsArray.map((key, index) => {
+    return `${pathAsArray.slice(0, index + 1).join('/')}/`; // ex: ['beep/', 'beep/boop/']. We need these model Ids to tell the LinkTo on the breadcrumb what to put into the dynamic *pathToSecret on the breadcrumb: ex/kv/beep/boop/
+  });
+
```

Comment:
it's a little hard to visualize what the result of this function is. Instead of the comment on line 20, it might be easier to update the comment on line 17 to say something like 
```js
/* 
takes a path "some/secret/path" and returns an array of objects with used for breadcrumbs: 
[
 { label: 'some', route: 'list-directory', model: 'some'},
 { label: 'secret', route: 'list-directory', model: 'some/secret/'},
 { label: 'path' ] } // last item is current route, only return label so breadcrumb has no link 
]
*/

Reply:
Good idea! I amended but replace some/secret/path with beep/boop/bop to keep consistent with tests and how we've referred to directories is other places in the app.